boxoffice/models/order.py

```diff
@@ -121,6 +121,12 @@ def confirmed_line_items(self):
 
         return LineItem.query.filter(LineItem.order == self, LineItem.status == LINE_ITEM_STATUS.CONFIRMED)
 
+    @property
+    def confirmed_cancelled_line_items(self):
+        from ..models import LineItem, LINE_ITEM_STATUS
+
+        return LineItem.query.filter(LineItem.order == self, LineItem.status.in_([LINE_ITEM_STATUS.CONFIRMED, LINE_ITEM_STATUS.CANCELLED]))
```

Comment:
From memory, this is what you need (please check for typos):

```python
confirmed_cancelled_line_items = db.relationship('LineItem', lazy='dynamic',
    primaryjoin='and_(LineItem.order_id == Order.id, LineItem.status.in_(' +
        repr([LINE_ITEM_STATUS.CONFIRMED, LINE_ITEM_STATUS.CANCELLED]) +
        '))')
```

Imports are not necessary as this relationship is being defined as a string (and hence the ugly repr hack). However, if this was defined as a backref from the LineItem model, we could use Python objects. @iambibhas, help Vidya with this.

Reply:
This will be cleaner if defined after the LineItem class:

```python
class LineItem(...):
    ...

Order.confirmed_cancelled_line_items = db.relationship(LineItem,
    lazy='dynamic',
    primaryjoin=db.and_(
        LineItem.order_id == Order.id,
        LineItem.status.in_([LINE_ITEM_STATUS.CONFIRMED, LINE_ITEM_STATUS.CANCELLED])
        )
    )
```

Also, the attribute is better off named as `confirmed_and_cancelled_line_items` as it's a little ambiguous without the `and`.